packages/@aws-cdk/aws-servicecatalog/lib/private/association-manager.ts

```diff
@@ -26,6 +27,31 @@ export class AssociationManager {
     };
   }
 
+  public static addEventNotifications(portfolio: IPortfolio, product: IProduct, topics: sns.ITopic[], options: TagUpdateConstraintOptions): void {
+    if (!topics.length) {
+      throw new Error(`No topics provided for launch notifications for association ${this.prettyPrintAssociation(portfolio, product)}`);
+    }
+    InputValidator.validateLength(this.prettyPrintAssociation(portfolio, product), 'description', 0, 2000, options.description);
+    const association = this.associateProductWithPortfolio(portfolio, product);
+
+    for (const topic of topics) {
+      const constructId = `LaunchNotificationConstraint${hashValues(topic.node.addr, association.associationKey)}`;
+      if (!portfolio.node.tryFindChild(constructId)) {
+        const constraint = new CfnLaunchNotificationConstraint(portfolio as unknown as cdk.Resource, constructId, {
+          acceptLanguage: options.messageLanguage,
+          description: options.description,
+          portfolioId: portfolio.portfolioId,
+          productId: product.productId,
+          notificationArns: [topic.topicArn],
+        });
+
+        // Add dependsOn to force proper order in deployment.
+        constraint.addDependsOn(association.cfnPortfolioProductAssociation);
+      } else {
+        throw new Error(`Topic ${topic} is already subscribed to association ${this.prettyPrintAssociation(portfolio, product)}`);
```

Comment:
We shouldn't just `.toString()` an entire `Topic` here - let's use something like `topic.node.path`.